### PR TITLE
[TEST] Increase Unit Test Coverage 

### PR DESCRIPTION
**Pull request by chiltonj**
Please review  @pedrofvteixeira @pdesai16 @niehusa @pentaho-jlien @pentaho-mracine

### CODE REVIEW

**Comment by rfellows** [comment]
@lucboudreau this test passes for both John and myself locally. are you aware of any environmental issues that could cause this on wingman?

File: extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java [hunk 1]
```diff
@@ -46,8 +56,29 @@ public void teardown() {
...
+    when( userSetting.getSettingValue() ).thenReturn( USER_SETTING_VALUE );
+
+    List<IUserSetting> userSettings = new ArrayList<>();
+    userSettings.add( userSetting );
+    when( userSettingService.getUserSettings() ).thenReturn( userSettings );
+
+    PentahoSystem.registerObject( userSettingService );
 
-    verify( userSettingService ).setUserSetting( "favorites", "[{\"fullPath\":\"/public/Steel Wheels/Top Customers (report).prpt\",\"title\":\"\\u003Cscript\\u003Eif(true\\u0026\\u0026\\u0027str\\u0027!=\\u0027str2\\u0027)alert(1)\\u003C/script\\u003E\",\"lastUse\":0}]" );
+    ArrayList<Setting> response = userSettingsResource.getUserSettings();
```
Comment (lucboudreau) [inline extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java]: This test does not actually test anything. You're mocking this method and invoking it here. It does not touch the actual class.

File: extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java [hunk 2]
```diff
@@ -46,8 +56,29 @@ public void teardown() {
...
+    assertNotNull( response );
+    assertTrue( !response.isEmpty() );
```
Comment (lucboudreau) [inline extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java]: This method belongs to a mock and is not mocked.

File: extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java [hunk 3]
```diff
@@ -46,8 +56,29 @@ public void teardown() {
...
+    assertTrue( response.get( 0 ).getName().equals( USER_SETTING_NAME ) );
+    assertTrue( response.get( 0 ).getValue().equals( USER_SETTING_VALUE ) );
```
Comment (lucboudreau) [inline extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java]: Same problem here and above. You're essentially testing mockito, not the class you intend to.

**Comment by lucboudreau** [comment]
@rfellows See my comments inline. Seems to me this test does not do what you think it does. Maybe I'm wrong, but that particular failure is because the method is not mocked so it returns whatever is the default.

File: extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java (hunk 3 above)
Comment (chiltonj) [inline extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java]: I still don't understand why the local runs pass but the Wingman's run doesn't?  I would like both environments to report same result.

File: extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java [hunk 4]
```diff
@@ -46,8 +56,29 @@ public void teardown() {
   public void testSetUserSetting() throws Exception {
     IUserSettingService userSettingService = mock( IUserSettingService.class );
     PentahoSystem.registerObject( userSettingService );
-    Response response = userSettingsResource.setUserSetting( "favorites", "[{\"fullPath\":\"/public/Steel Wheels/Top Customers (report).prpt\", \"title\":\"<script>if(true&&'str'!='str2')alert(1)</script>\", \"lastUse\":0}]" );
+    Response response = userSettingsResource.setUserSetting( USER_SETTING_NAME, "[{\"fullPath\":\"/public/Steel Wheels/Top Customers (report).prpt\", \"title\":\"<script>if" + "(true&&'str'!='str2')alert(1)</script>\", \"lastUse\":0}]" );
+
+    verify( userSettingService ).setUserSetting( USER_SETTING_NAME,      USER_SETTING_VALUE );
```
Comment (lucboudreau) [inline extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java]: Not sure what's tested here either. You called a mock with an unmocked method, then verifying that you indeed called it one line earlier?

File: extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java (hunk 3 above)
Comment (lucboudreau) [inline extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java]: Try the exact same commands as above.

```
ant -Dpentaho.resolve.repo=http://ivy-nexus.pentaho.org/content/groups/approved -Dtestreports.dir=bin/reports/unit-test -f extensions/build.xml clean-all resolve jacoco 
```

```
ant -Dpentaho.resolve.repo=http://ivy-nexus.pentaho.org/content/groups/approved -Dtestreports.dir=bin/reports/integration-test -f extensions/build.xml jacoco-integration checkstyle publish-local
```

File: extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java (hunk 1 above)
Comment (chiltonj) [inline extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java]: I am not sure I follow: userSettingsResource.getUserSettings() is not a mocked method.

Comment (lucboudreau) [inline extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java]: Hey John, my bad. I didn't catch the difference between *Resource and *Service.

**Comment by pedrofvteixeira** [comment]
Cherry-picked onto my local machine, all tests run successfully. 
Wingman is complaining of a coverage drop in a MondrianConfigProperties class, which has no relation to the unit tests that were made. 

PR is good to go from my part, but in light of that coverage drop I'll defer to @lucboudreau for the final say.

**Pull request merged**